src/plugins/intel_gpu/src/graph/impls/ocl/non_max_suppression.cpp

```diff
@@ -60,8 +60,9 @@ struct non_max_suppression_impl : typed_primitive_impl_ocl<non_max_suppression>
     }
 
 public:
-    static std::unique_ptr<primitive_impl> create(const non_max_suppression_node& arg, const kernel_impl_params& impl_param) {
+static kernel_params_t get_kernel_params(const kernel_impl_params& impl_param, bool is_shape_agnostic = false) {
         const auto& primitive = impl_param.typed_desc<non_max_suppression>();
+        const auto& arg = impl_param.prog->get_node(impl_param.desc->id).as<non_max_suppression>();
```

Comment:
Perhaps all required from program_node properties could be expressed using primitive? And then we could get rid of program_node dependency here at all
But I'm okay with current solution too